tools/bumper/cnaoRepoCommands.go

```diff
@@ -0,0 +1,168 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/coreos/go-semver/semver"
+	"github.com/go-git/go-git/v5"
+	"github.com/pkg/errors"
+)
+
+type gitCnaoRepo struct {
+	configParams *component
+
+	githubInterface githubInterface
+
+	gitRepo *gitRepo
+}
+
+const (
+	repoUrl = "https://github.com/kubevirt/cluster-network-addons-operator"
+)
+
+func getCnaoRepo(api *githubApi) (*gitCnaoRepo, error) {
+	cnaoGitRepo, err := openGitRepo(".")
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to get git repo for cnao repo")
+	}
+
+	cnaoComponentParams := &component{
+		Url: repoUrl,
+	}
+
+	gitCnaoRepo := &gitCnaoRepo{
+		configParams:    cnaoComponentParams,
+		githubInterface: api,
+		gitRepo:         cnaoGitRepo,
+	}
+
+	return gitCnaoRepo, nil
+}
+
+// openGitRepo opens an existing repository
+func openGitRepo(repoPath string) (*gitRepo, error) {
+	repo, err := git.PlainOpen(repoPath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to open existing repo in path %s", repoPath)
+	}
+
+	return &gitRepo{
+		repo:     repo,
+		localDir: repoPath,
+	}, nil
+}
+
+func (cnaoRepoOps *gitCnaoRepo) getComponentsConfig(relativeConfigPath string) (componentsConfig, error) {
+	configPath := filepath.Join(cnaoRepoOps.gitRepo.localDir, relativeConfigPath)
+	return parseComponentsYaml(configPath)
+}
+
+func (cnaoRepoOps *gitCnaoRepo) updateComponentsConfig(relativeConfigPath string, componentsConfig componentsConfig) error {
+	configPath := filepath.Join(cnaoRepoOps.gitRepo.localDir, relativeConfigPath)
+	return updateComponentsYaml(configPath, componentsConfig)
+}
+
+func (cnaoRepoOps *gitCnaoRepo) bumpComponent(componentName string) error {
+	logger.Printf("Running bump-%s script", componentName)
+	cmd := exec.Command("make", "-C", cnaoRepoOps.gitRepo.localDir, fmt.Sprintf("bump-%s", componentName))
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	err := cmd.Run()
+	if err != nil {
+		return errors.Wrapf(err, "Failed to run bump script, \nStdout:\n%s\nStderr:\n%s", cmd.Stdout, cmd.Stderr)
+	}
+	return nil
+}
+
+func (cnaoRepoOps *gitCnaoRepo) isComponentBumpNeeded(currentReleaseVersion, latestReleaseVersion, updatePolicy, proposedPrTitle string) (bool, error) {
+	logger.Printf("currentReleaseVersion: %s, latestReleaseVersion: %s, updatePolicy: %s\n", currentReleaseVersion, latestReleaseVersion, updatePolicy)
+
+	if updatePolicy == updatePolicyStatic {
+		logger.Printf("updatePolicy is static. avoiding auto bump")
+		return false, nil
+	}
+
+	// check if PR not already opened
+	isAlreadyOpened, err := cnaoRepoOps.isPrAlreadyOpened(cnaoRepoOps.getCnaoRepoOwnerFromUrl(), cnaoRepoOps.getCnaoRepoNameFromUrl(), proposedPrTitle)
```

Comment:
We can reduce this to `cnaoRepoOps.isPrAlreadyOpened(proposedPrTitle)`

Reply:
Done